Library/Homebrew/tap.rb

```diff
@@ -63,13 +63,16 @@ def self.fetch(*args)
 
   def self.from_path(path)
     match = File.expand_path(path).match(HOMEBREW_TAP_PATH_REGEX)
-    return if match.blank? || match[:user].blank? || match[:repo].blank?
 
-    fetch(match[:user], match[:repo])
+    return unless match
+    return unless (user = match[:user])
+    return unless (repo = match[:repo])
+
+    fetch(user, repo)
   end
 
   # @private
-  sig { params(name: String).returns(T.nilable([T.attached_class, String])) }
+  sig { params(name: String).returns(T.nilable([Tap, String])) }
   def self.with_formula_name(name)
```

Comment:
Why does this need to change?

Reply:
Because it itself uses `fetch`, so can only return `Tap`.